src/repl/mod.rs

```diff
@@ -70,16 +70,12 @@ pub struct REPLImpl {
     cache: Cache,
     /// The parser, supporting toplevel let declaration.
     parser: grammar::ExtendedTermParser,
-    /// The eval environment. Contain the global environment with the stdlib, plus toplevel
+    /// The global environment. Contain the global environment with the stdlib, plus toplevel
     /// declarations and loadings made inside the REPL.
-    eval_env: eval::Environment,
-    /// The initial eval environment, without the toplevel declarations made inside the REPL. Used
+    env: GlobalEnv,
```

Comment:
I'm not sure we should `GlobalEnv` here. `GlobalEnv` is used to return a pair of environment from the cache, but here I feel it's just adding an indirection that is not very useful. We could inline `eval_env` and `type_env` as before. But it's just a nitpicking suggestion, feel free to ignore.

Reply:
I don't know, I did it because:
- It keep both mutable envs grouped (separated from the init_type_env)
- It keep the `new` function simple (only the object construction and return without any variable declaration).
- I thought the indirection is small and I suppose well optimized by Rustc.

so I will let it as it is for now and if it's an issue at one point, it can be revert easily.